insurance.go

```diff
@@ -9,33 +9,33 @@ import (
 // EasyPost API and shipments purchased through third parties and later
 // registered with EasyPost.
 type Insurance struct {
-	ID           string    `json:"id,omitempty"`
-	Object       string    `json:"object,omitempty"`
-	Reference    string    `json:"reference,omitempty"`
-	Mode         string    `json:"mode,omitempty"`
-	CreatedAt    *DateTime `json:"created_at,omitempty"`
-	UpdatedAt    *DateTime `json:"updated_at,omitempty"`
-	Amount       string    `json:"amount,omitempty"`
-	Carrier      string    `json:"carrier,omitempty"`
-	Provider     string    `json:"provider,omitempty"`
-	ProviderID   string    `json:"provider_id,omitempty"`
-	ShipmentID   string    `json:"shipment_id,omitempty"`
-	TrackingCode string    `json:"tracking_code,omitempty"`
-	Status       string    `json:"status,omitempty"`
-	Tracker      *Tracker  `json:"tracker,omitempty"`
-	ToAddress    *Address  `json:"to_address,omitempty"`
-	FromAddress  *Address  `json:"from_address,omitempty"`
-	Fee          *Fee      `json:"fee,omitempty"`
-	Messages     []string  `json:"messages,omitempty"`
+	ID           string    `json:"id,omitempty" url:"id,omitempty"`
```

Comment:
It doesn't appear as if you can supply a function for a [struct field tag](https://www.dolthub.com/blog/2024-02-07-go-tags/), it needs to be a static string during compile time.

We could potentially [implement our own struct field tag](https://medium.com/pragmatic-programmers/parsing-struct-tags-a99c5db9accd) that would under-the-hood handle both URL and JSON encoding, but I believe it's best to simply list both the [existing defined tags](https://go.dev/wiki/Well-known-struct-tags) rather than reinvent the wheel.

Reply:
Dang, too bad. Thanks for looking!